### PR TITLE
initial commit for issue-15

### PR DESCRIPTION
**Pull request by rajeshkio**
I am opening this PR for review and some ideas regarding my issue.
With the current implementation, I can find and choose one of the cheapest metros, but I am getting this error:

 Error: POST https://api.equinix.com/metal/v1/projects/96810823-2102-43b1-a122-46f6dff73a01/spot-market-requests?include=devices%2Cproject%2Cplan: 422 No available servers with plan m3.small.x86 in facility am2 
│ 
│   with equinix_metal_spot_market_request.seed_spot_request[0],
│   on main.tf line 70, in resource "equinix_metal_spot_market_request" "seed_spot_request":
│   70: resource "equinix_metal_spot_market_request" "seed_spot_request" {
│ 

Another solution would be to keep trying all cheap metros till this is successfully deployed.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -10,9 +10,46 @@ resource "random_password" "token" {
   special = false
 }
 
+// Find the cheapest metro
+
+provider "http" {}
```
Comment (dnoland1) [inline main.tf:15]: Can you move the provider block to provider.tf?

File: main.tf [hunk 2]
```diff
@@ -10,9 +10,46 @@ resource "random_password" "token" {
...
+
+data "http" "prices" {
+  url    = "https://api.equinix.com/metal/v1/market/spot/prices/metros"
+  method = "GET"
+  request_headers = {
+    "X-Auth-Token" = var.api_key
```
Comment (dnoland1) [inline main.tf:21]: Can you move the data block to data.tf?

**Comment by dnoland1** [comment]
I'm checking with Equinix support on why it's choosing a facility within the metro that does not have servers available.

**Comment by rajeshkio** [comment]
Most of the metro have removed 0.11 spot pricing, and several of them shows no servers available when you try to create a server.

equinix_metal_spot_market_request.seed_spot_request[0]: Creating...
╷
│ Error: POST https://api.equinix.com/metal/v1/projects/96810823-2102-43b1-a122-46f6dff73a01/spot-market-requests?include=devices%2Cproject%2Cplan: 422 No available servers with plan m3.small.x86 in facility sy5 
│ 
│   with equinix_metal_spot_market_request.seed_spot_request[0],
│   on main.tf line 33, in resource "equinix_metal_spot_market_request" "seed_spot_request":
│   33: resource "equinix_metal_spot_market_request" "seed_spot_request" {
│ 
╵
╷
│ Error: POST https://api.equinix.com/metal/v1/projects/96810823-2102-43b1-a122-46f6dff73a01/spot-market-requests?include=devices%2Cproject%2Cplan: 422 No available servers with plan m3.small.x86 in facility sy5 
│ 
│   with equinix_metal_spot_market_request.join_spot_request[0],
│   on main.tf line 71, in resource "equinix_metal_spot_market_request" "join_spot_request":
│   71: resource "equinix_metal_spot_market_request" "join_spot_request" {
│

**Comment by dnoland1** [comment]
@rajeshkio `terraform plan` is failing in drone. Can you have a look, see https://drone-pr.rancher.io/rancherlabs/harvester-equinix-terraform/19/1/4 ?

**Comment by rajeshkio** [comment]
> @rajeshkio `terraform plan` is failing in drone. Can you have a look, see https://drone-pr.rancher.io/rancherlabs/harvester-equinix-terraform/19/1/4 ?

okay, it seems I need to update ..droney.yaml with the new environment variable

**Review by dnoland1 — CHANGES_REQUESTED** [review]
Tested and works fine. There are three files that have a merge conflict, can you take a look?

Also, we want to retain the ability for users to select a metro and min price. Perhaps we can have a boolean variable like `use_cheapest_metro`. If true, use the new logic, if false, should behave as before this change.

**Review by dnoland1 — CHANGES_REQUESTED** [review]
It looks like the merge reversed various changes in the main.tf for the project_id. Can you take a look?

**Comment by rajeshkio** [comment]
> There are three files that have a merge

One simple solution is to use this variable and add checks while choosing metro and max_bid_price. But this doesn't stop the curl request in data.tf. I am thinking when we do not want to use the cheapest metro we shouldn't do the curl request as well.

**Comment by rajeshkio** [comment]
> Tested and works fine. There are three files that have a merge conflict, can you take a look?
> 
> Also, we want to retain the ability for users to select a metro and min price. Perhaps we can have a boolean variable like `use_cheapest_metro`. If true, use the new logic, if false, should behave as before this change.

One simple solution will be to add use_cheapest_metro boolean in my existing code and add the check while choosing metro and max_bid_price. But I am thinking even when use_cheapest_metro is false, the curl request in data.tf will still run. That's an extra process, thinking how to change that.

**Comment by rajeshkio** [comment]
scraping this PR in favor of https://github.com/rancherlabs/harvester-equinix-terraform/pull/63

**Pull request closed without merging**